src/ChangeStream/ChangeStream.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\ChangeStream;
+
+use Puli\Repository\Api\Resource\PuliResource;
+use Puli\Repository\Api\ResourceRepository;
+
+/**
+ * Stream to track repositories changes and fetch previous versions of resources.
+ *
+ * @since  1.0
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class ChangeStream
```

Comment:
IIUC this is not yet persistent right? IMO `ChangeStream` should be an interface in `Api`, and this one is probably an `InMemoryChangeStream`, no?